code/modules/mob/mob_movement.dm

```diff
@@ -200,8 +200,15 @@
 							M.animate_movement = 2
 							return
 
-		if(mob.confused && IsEven(world.time))
-			step(mob, pick(cardinal))
+		if(mob.confused)
+			if(mob.confused > 100)
+				step(mob, pick(cardinal))
+			else if(prob(mob.confused / 2))

```

Comment:
nitpick, replace ```/ 2``` with ```* 0.5``` mathematically the same, but faster for code to execute.